lib/kafo/progress_bar.rb

```diff
@@ -11,7 +11,7 @@ module Kafo
 # change more methods like #done_message or #print_error
   class ProgressBar
     MONITOR_RESOURCE = %r{\w*MONITOR_RESOURCE ([^\]]+\])}
-    EVALTRACE_START = %r{/(.+\]): Starting to evaluate the resource}
+    EVALTRACE_START = %r{/(.+\]): Starting to evaluate the resource( \((\d+) of (\d+)\))?}
```

Comment:
Back when I wrote the initial version we couldn't use capture groups in regexes because we needed to support Ruby 2.0. Now we can and that makes the code below a lot easier:
```ruby
regex = %r{/(?<resource>.+\]): Starting to evaluate the resource( \((?<number>\d+) of (?<total>\d+)\))?}
match = regex.match('/Stage[main]/Example/File[/foo/bar]: Starting to evaluate the resource (1 of 2)')
assert match[:resource] == 'Stage[main]/Example/File[/foo/bar]'
assert match[:number] == '1'
assert match[:total] == '2'

legacy_match = regex.match('/Stage[main]/Example/File[/foo/bar]: Starting to evaluate the resource')
assert legacy_match[:resource] == 'Stage[main]/Example/File[/foo/bar]'
assert legacy_match[:number].nil?
assert legacy_match[:total].nil?
```

Reply:
Switched for each resource to this method. This is much cleaner to read, thanks for pointing it out.